ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/customercenter/viewmodel/CustomerCenterViewModel.kt

```diff
@@ -93,12 +100,16 @@ internal class CustomerCenterViewModelImpl(
     }
 
     override fun dismissRestoreDialog() {
-        val currentState = _state.value
-        if (currentState is CustomerCenterState.Success) {
-            _state.value = currentState.copy(
-                showRestoreDialog = false,
-                restorePurchasesState = RestorePurchasesState.INITIAL,
-            )
+        _state.update {
+            val currentState = _state.value
```

Comment:
`update` already receives the current value: 
```suggestion
        _state.update { currentState ->
```

Reply:
Oh I see this applies to a few more lines in this class.